### PR TITLE
feat(rate-limiting) implement rate limiting by path. 

### PR DESCRIPTION
**Pull request by kikito**
This PR expands #6234 (by @fairyqb) with a couple integration tests as well as changing the way the ratelimiting is calculated (in the original PR `conf.path` was just ignored by `kong.request.get_path`).

Left my changes as separate commits for easier review, but this PR should be Squashed and Merged into a single commit when approved.

Closes #6234. Closes #6202

### CODE REVIEW

**Comment by fairyqb** [comment]
title right ?

File: kong/plugins/rate-limiting/handler.lua [hunk 1]
```diff
@@ -67,6 +67,12 @@ local function get_identifier(conf)
 
   elseif conf.limit_by == "header" then
     identifier = kong.request.get_header(conf.header_name)
+
+  elseif conf.limit_by == "path" then
+    local req_path = kong.request.get_path()
+    if req_path == conf.path then
+      identifier = kong.request.get_path()
```
Comment (gszr) [inline kong/plugins/rate-limiting/handler.lua:74]:
```suggestion
      identifier = req_path
```

File: kong/plugins/rate-limiting/handler.lua [hunk 2]
```diff
@@ -67,6 +67,12 @@ local function get_identifier(conf)
 
   elseif conf.limit_by == "header" then
     identifier = kong.request.get_header(conf.header_name)
+
+  elseif conf.limit_by == "path" then
+    local req_path = kong.request.get_path()
+    if req_path == conf.path then
+      identifier = req_path
+    end
```
Comment (fairyqb) [inline kong/plugins/rate-limiting/handler.lua:75]: Whether it is necessary to ignore case ?

**Comment by fairyqb** [comment]
1. consumer and service should be “logic and” relationships.
2. credential and service should be “logic and” relationships.

File: kong/plugins/rate-limiting/handler.lua (hunk 2 above)
Comment (kikito) [reply kong/plugins/rate-limiting/handler.lua:75]: While some websites/services might decide to make their urls case-insensitive, the [html standard considers them sensitive](https://www.w3.org/TR/WD-html40-970708/htmlweb.html), so `/foo` and `/FOO`, in principle, point to different resources.

Case-sensitivity has a cluster of complexity in dealing with non-English letters - for example the Cyrillic and Greek alphabets have cased letters which can be used in paths, with encodings (see https://www.w3.org/International/articles/idn-and-iri/#path).

File: kong/plugins/rate-limiting/handler.lua [hunk 3]
```diff
@@ -67,6 +67,12 @@ local function get_identifier(conf)
 
   elseif conf.limit_by == "header" then
     identifier = kong.request.get_header(conf.header_name)
```
Comment (fairyqb) [inline kong/plugins/rate-limiting/handler.lua:69]: Only header names have fewer scenarios, and more headers and values, I think.

**Pull request merged**